packages/fx-core/src/common/local/taskDefinition.ts

```diff
@@ -0,0 +1,180 @@
+        : `start --javascript --language-worker="--inspect=9229" --port "7071" --cors "*"`;
+    // TODO: get command path from function checker
+    // const funcChecker = new FuncToolChecker();
+    return {
+      name: "backend start",
+      command: "", // TODO: `${await funcChecker.getFuncCommand()} ${args}`
+      env: undefined, // TODO: get local env
+      cwd: path.join(projectRoot, FolderName.Function),
+      isShell: true,
+      isBackground: true,
+    };
+  }
+
+  static async getBackendWatchTaskDefinition(projectRoot: string): Promise<ITaskDefinition> {
+    return {
+      name: "backend watch",
+      command: "npx tsc --watch",
+      env: undefined, // TODO: get local env
+      cwd: path.join(projectRoot, FolderName.Function),
+      isShell: true,
+      isBackground: true,
+    };
+  }
+
+  static async getAuthTaskDefinition(localSettings: Json): Promise<ITaskDefinition> {
+    // TODO: get command path from dotnet checker
+    // const dotnetChecker = new DotnetChecker();
+
+    return {
+      name: "auth start",
+      command: "", // TODO: await dotnetChecker.getDotnetExecPath()
+      args: ["Microsoft.TeamsFx.SimpleAuth.dll"],
+      env: undefined, // TODO: get local env
+      cwd: localSettings.auth.simpleAuthFilePath,
+      isShell: false,
+      isBackground: true,
+    };
+  }
+
+  static async getBotTaskDefinition(
+    projectRoot: string,
+    projectSettings: ProjectSettings
+  ): Promise<ITaskDefinition> {
+    const command =
+      projectSettings.programmingLanguage === ProgrammingLanguage.typescript
+        ? "npx nodemon --exec node --inspect=9239 --signal SIGINT -r ts-node/register index.ts"
+        : "npx nodemon --inspect=9239 --signal SIGINT index.js";
+    return {
+      name: "bot start",
+      command: command,
+      env: undefined, // TODO: get local env
+      cwd: path.join(projectRoot, FolderName.Bot),
+      isShell: true,
+      isBackground: true,
+    };
+  }
+
+  static async getNgrokTaskDefinition(
+    projectRoot: string,
+    localSettings: Json
+  ): Promise<ITaskDefinition> {
+    // TODO: use ngrok checker to get ngrok bin folder
+    // const ngrokChecker = new NgrokChecker();
+    // const ngrokBinFolder = ngrokChecker.getNgrokBinFolder();
+    const skipNgrok = (localSettings?.bot?.skipNgrok as boolean) === true;
+    return {
+      name: "ngrok start",
+      command: skipNgrok
+        ? "echo 'Do not start ngrok, but use predefined bot endpoint.'"
+        : "npx ngrok http 3978 --log=stdout",
+      env: {
+        PATH: undefined, // TODO: `${ngrokBinFolder}${path.delimiter}${process.env.PATH ?? ""}`,
+      },
+      cwd: path.join(projectRoot, FolderName.Bot),
+      isShell: true,
+      isBackground: !skipNgrok,
+    };
+  }
+
+  static getFrontendInstallTaskDefinition(projectRoot: string): ITaskDefinition {
+    return {
+      name: "frontend npm install",
+      command: "npm install",
```

Comment:
Also add `--no-audit` as mentioned in https://github.com/OfficeDev/TeamsFx/pull/3469, thanks~